lib/FrontendTool/FrontendTool.cpp

```diff
@@ -604,24 +604,30 @@ constructDetailedTaskDescription(const CompilerInvocation &Invocation,
     CommandLine += std::string(" ") + A;
   }
 
-  // Primary Input only
-  Inputs.push_back(CommandInput(PrimaryInput.getFileName()));
+  // Primary Inputs
+  for (const auto &input : PrimaryInputs) {
+    Inputs.push_back(CommandInput(input.getFileName()));
+  }
 
-  // Output for this Primary
-  auto OutputFile = PrimaryInput.outputFilename();
-  Outputs.push_back(OutputPair(file_types::lookupTypeForExtension(
-                                   llvm::sys::path::extension(OutputFile)),
-                               OutputFile));
+  // Outputs for Primary Inputs
+  for (const auto &input : PrimaryInputs) {
+    auto OutputFile = input.outputFilename();
+    Outputs.push_back(OutputPair(file_types::lookupTypeForExtension(
+                                                                    llvm::sys::path::extension(OutputFile)),
+                                 OutputFile));
+  }
```

Comment:
I believe this will be equivalent to what the code does now already. Right below this code-block we iterate over all the inputs again and gather supplementary outputs per-input. So while the iteration may be redundant, we should not miss any outputs. I've updated the test with an additional supplementary output just to make sure it is consistently shown in both WMO and non-WMO.

Reply:
Right yeah, we won't miss any outputs, but I believe we'll be writing out empty outputs for all but the first input in a WMO invocation with multiple inputs and a single main output. Is that expected?

Reply:
Ah, of course, now I see it. Will fix, thank you.